pkg/cloudprovider/provider/vsphere/provider.go

```diff
@@ -308,14 +312,18 @@ func (p *provider) Create(machine *v1alpha1.Machine, _ cloud.MachineUpdater, use
 	if err != nil {
 		return nil, fmt.Errorf("failed to get vsphere client: '%v'", err)
 	}
-	defer client.Logout(context.TODO())
+	defer func() {
+		if lerr := client.Logout(context.TODO()); lerr != nil {
+			glog.Errorf("vsphere client failed to logout: %s", lerr)
```

Comment:
Instead of logging those situations, we should call `utilruntime.HandleErr`
It also logs the error, but additionally increases the error-counter-metric

Reply:
Fixed